src/core/mac/mac.cpp

```diff
@@ -1540,21 +1538,32 @@ void Mac::ReceiveDoneTask(Frame *aFrame, otError aError)
         ExitNow(error = OT_ERROR_INVALID_SOURCE_ADDRESS);
     }
 
-    // Source Whitelist Processing
-    if (srcaddr.mLength != 0 && mWhitelist.IsEnabled())
+    // Source filter Processing
+    if (srcaddr.mLength != 0)
     {
-        VerifyOrExit((whitelistEntry = mWhitelist.Find(srcaddr.mExtAddress)) != NULL, error = OT_ERROR_WHITELIST_FILTERED);
+        rssi = mFilter.RssiInFilterGet();
```

Comment:
_Suggestion_: What if we add a new method to `Filter` class to contain the filtering code here. This helps encapsulate the logic of filter state (whitelisting vs blacklisting) inside the `Filter` class itself and keeps the `Mac` code easier to read. 

For example, we can define a method 
`otError Filter::Apply(const Mac::ExtAddress &aExtAddress, int8_t &aRss)`
which returns
- `OT_ERROR_NONE`: Address is not filtered and ok to continue.
- `OT_ERROR_WHITELIST_FILTERED` or `OT_ERROR_BLACKLIST_FILTERED`  if the frame should be filtered

and updates the `aRss` value (if requried). 

The code in `Mac` would be: 
````
if (srcaddr.mLength != 0) 
{
    SuccessOrExit(error = mFilter.Apply(srcaddr.mExtAddress, aFrame->mPower));
}
````